pages/desktop/consumer_pages/base.py

```diff
@@ -84,6 +86,29 @@ class HeaderRegion(Page):
         def is_user_logged_in(self):
             return self.is_element_visible(*self._settings_toggle_locator)
 
+        @property
+        def categories_name(self):
+            return self.selenium.find_element(*self._categories_toggle_locator).text
+
+        def open_categories_menu(self):
+            categories = self.selenium.find_element(By.ID, 'header--categories')
+            self.selenium.find_element(*self._categories_toggle_locator).click()
+            WebDriverWait(self.selenium, self.timeout).until(self.element_settled(categories))
+
+        class element_settled(object):
+
+            def __init__(self, element):
+                self.el = element
+                self.state = []
+
+            def __call__(self, selenium):
+                self.state.append('{0.location} {0.size}'.format(self.el))
+                return len(self.state) >= 3 and len(set(self.state[-3:])) == 1
+
+        @property
+        def categories(self):
+            return [self.Category(self.testsetup, element) for element in self.selenium.find_elements(*self._categories_locator)]
```

Comment:
I find lines like this difficult to break in a way that I am satisfied, but I'll do something about it.